securesystemslib/signer/_aws_signer.py

```diff
@@ -101,136 +116,119 @@ def import_(cls, aws_key_id: str, local_scheme: str) -> Tuple[str, Key]:
         Raises:
             UnsupportedAlgorithmError: If the AWS KMS signing algorithm is
             unsupported.
-            BotoCoreError: Errors from the botocore.exceptions library.
+            BotoCoreError: Errors from the botocore library.
             ClientError: Errors related to AWS KMS client.
         """
         if AWS_IMPORT_ERROR:
             raise UnsupportedLibraryError(AWS_IMPORT_ERROR)
 
-        client = boto3.client("kms")
-        request = client.get_public_key(KeyId=aws_key_id)
+        try:
+            if local_scheme and local_scheme in cls.aws_signing_algorithms:
+                scheme = local_scheme
+        except KeyError as e:
+            raise ValueError(f"Unsupported scheme: {local_scheme}") from e
```

Comment:
You are mixing EAFP and LBYL here with the consequence that your ValueError will never be raised. :) 

I suggest to write:

```suggestion
        if local_scheme:
            if local_scheme not in cls.aws_signing_algorithms:
                raise ValueError(f"Unsupported scheme: {local_scheme}") from e
                
        # no need to rename `local_scheme`, you can just use this name below
```

Reply:
ah, thank you.

Reply:
Very welcome!

Sorry if my comment about `local_scheme` was unclear. If you don't rename it here, you will need to use `local_scheme` below. See my code suggestions...